### PR TITLE
Add experimental custom pragma

### PR DESCRIPTION
**Pull request by jxnblk**
Trying out the idea of using a custom pragma for the Styled System `css` call. This uses Emotion's `jsx` create element function and is intended to be a replacement for that. I've use the `@jsx` pragma call in the docs to show explicitly what's happening and disabled the emotion plugin. Not sure if this is worth exploring more but wanted to give it a shot cc @johno @ChristopherBiscardi

### CODE REVIEW

**Comment by johno** [comment]
🔥🔥🔥

This is super interesting and I think being able to remove the `css={css({` is a substantial improvement in DX.

File: theme-ui/index.js [hunk 1]
```diff
@@ -1,14 +1,72 @@
...
+const compose = (...funcs) => (type, props, ...children) => {
+  return funcs.reduce((el, fn) => {
+    return fn.apply(undefined, [ el.type, el.props, ...children ])
+  }, React.createElement.apply(undefined, [ type, props, ...children ]))
+}
+
+export const jsx = compose(
+  system,
+  emotion
+)
```
Comment (jxnblk) [inline theme-ui/index.js]: FWIW I tried throwing the `mdx` create element function in here as well, and it *seemed* to work as expected, replacing e.g. `<h1>` with a component from context – something to think about as an alternative to the `Styled.h1` components

**Comment by ChristopherBiscardi** [comment]
Very promising direction IMO (sidenote: we've now all written custom pragmas for themes'mdx stuff lol).

File: theme-ui/index.js (hunk 1 above)
Comment (ChristopherBiscardi) [reply theme-ui/index.js]: That is really awesome. We should put some thought into using the mdx createElement function more generally (even though it seems to work).

**Pull request merged**